RabbitMQ.Stream.Client/RawProducer.cs

```diff
@@ -150,17 +150,19 @@ public async ValueTask Send(ulong publishingId, List<Message> subEntryMessages,
         {
             if (subEntryMessages.Count != 0)
             {
-                await SemaphoreWait();
+                await SemaphoreAwaitAsync();
                 var publishTask =
                     _client.Publish(new SubEntryPublish(_publisherId, publishingId,
                         CompressionHelper.Compress(subEntryMessages, compressionType)));
-                if (!publishTask.IsCompletedSuccessfully)
```

Comment:
Changed due of https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2012